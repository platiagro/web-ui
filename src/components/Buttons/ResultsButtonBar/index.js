import ResultsButtonBar from "./ResultsButtonBar.component";

export default ResultsButtonBar;
