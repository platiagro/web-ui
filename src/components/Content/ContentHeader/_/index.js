// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { PageHeader } from 'antd';
import { ArrowLeftOutlined } from '@ant-design/icons';

// COMPONENTS
import Title from '../../Title';

// STYLES
import './style.less';

/**
 * Content Header.
 * This component is responsible for displaying the content header with editable
 * title and go back arrow.
 *
 * @param props
 */
const ContentHeader = (props) => {
  // destructuring props
  const {
    handleGoBack,
    handleSubmit,
    title,
    loading,
    subTitle,
    extra,
    backIcon,
  } = props;

  // RENDER
  return (
    // page header component
    <PageHeader
      className='contentHeader'
      title={
        // title component
        <>
          <span className='subtitle-custom'>{subTitle}</span>
          <Title
            loading={loading}
            level={3}
            title={title}
            handleSubmit={handleSubmit}
          />
        </>
      }
      onBack={handleGoBack}
      extra={extra}
      backIcon={backIcon}
    />
  );
};

// PROP TYPES
ContentHeader.propTypes = {
  /** content header go back function */
  handleGoBack: PropTypes.func.isRequired,
  /** content header edit submit function */
  handleSubmit: PropTypes.func,
  /** content header title */
  title: PropTypes.string.isRequired,
  /** content header is loading */
  loading: PropTypes.bool.isRequired,
  /** if false the back icon will not be displayed */
  backIcon: PropTypes.bool.isRequired,
};

// PROP DEFAULT VALUES
ContentHeader.defaultProps = {
  /** content header edit submit function */
  handleSubmit: undefined,
  /* show back icon */
  backIcon: <ArrowLeftOutlined />,
};

// EXPORT
export default ContentHeader;
