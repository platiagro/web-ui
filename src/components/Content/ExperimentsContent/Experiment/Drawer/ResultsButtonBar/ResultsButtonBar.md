**Default**

```js
const [loading, setLoading] = React.useState(false);

const onClick = () => {
  console.log("clicked");
  setLoading(true);
  setTimeout(() => {
    setLoading(false);
  }, 2500);
};

<ResultsButtonBar
  disabled={false}
  loading={loading}
  showingResults={false}
  handleEditClick={onClick}
  handleResultsClick={onClick}
/>;
```
