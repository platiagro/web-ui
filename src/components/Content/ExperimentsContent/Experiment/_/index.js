import React from 'react';

import { OperatorResultsModalContainer } from 'containers';

import ExperimentFlowContainer from '../ExperimentFlow/_/ExperimentFlowContainer';

const Experiment = () => {
  return (
    <>
      <OperatorResultsModalContainer />
      <ExperimentFlowContainer />;
    </>
  );
};

export default Experiment;
