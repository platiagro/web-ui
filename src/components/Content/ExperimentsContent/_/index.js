// CORE LIBS
import React from 'react';
import { useParams } from 'react-router-dom';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';

// COMPONENTS
import TasksMenuBlock from '../TasksMenuBlock/_/Container';
import ExperimentsTabs from '../ExperimentsTabs/_/Container';
import NewExperimentButton from '../NewExperimentButton/Container';
import NewExperimentModal from '../NewExperimentModal/Container';

import {
  ChangeRoutePromptContainer,
  CompareResultsModalContainer,
  DataViewModalContainer,
  HeaderExperimentsContentContainer,
  OperatorResizableSectionContainer,
  OperatorResultsModalContainer,
} from 'containers';

import FlowDrop from './FlowDrop';

import { Layout } from 'antd';
import './style.less';

const { Footer, Sider, Content } = Layout;

const ExperimentsContent = () => {
  const { experimentId } = useParams();

  const renderFlowContent = () => (
    <Layout style={{ overflow: 'hidden' }}>
      <Sider width={250}>
        <TasksMenuBlock disabled={!experimentId} />
      </Sider>
      <Content>
        <Layout style={{ height: '100%' }}>
          <Content style={{ display: 'flex' }}>
            <FlowDrop />
            <OperatorResizableSectionContainer />
          </Content>
          <Footer style={{ padding: 0 }}>
            <div className='custom-experiment-tabs'>
              <ExperimentsTabs />
              <NewExperimentButton />
              <NewExperimentModal />
            </div>
          </Footer>
        </Layout>
      </Content>
    </Layout>
  );

  // RENDER
  return (
    <>
      <DndProvider backend={HTML5Backend}>
        <CompareResultsModalContainer />
        <DataViewModalContainer />
        <OperatorResultsModalContainer />
        <HeaderExperimentsContentContainer />
        {renderFlowContent()}
        <ChangeRoutePromptContainer />
      </DndProvider>
    </>
  );
};

// EXPORT
export default ExperimentsContent;
