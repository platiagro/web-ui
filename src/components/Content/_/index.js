// CORE LIBS
import React from 'react';
import { Switch, Route } from 'react-router-dom';

// UI LIBS
import { Layout } from 'antd';

// CONTENTS
import ProjectsContent from '../ProjectsContent/_';
import ProjectContent from '../ProjectContent/_';
import TaskContent from '../TasksContent/_';
import ImplantedExperimentsContent from '../ImplantedExperimentsContent/_';
import Error404Content from '../Error404Content';

// STYLES
import './style.scss';

/**
 * Content.
 * This component is responsible for displaying the content.
 */
const Content = () => {
  // RENDER
  return (
    // layout component
    <Layout>
      <Switch>
        {/* projects content */}
        <Route exact path='/'>
          <ProjectsContent />
        </Route>
        {/* projects content */}
        <Route exact path='/projetos'>
          <ProjectsContent />
        </Route>
        {/* project content */}
        <Route exact path='/projetos/:projectId'>
          <ProjectContent />
        </Route>
        {/* experiment content */}
        <Route exact path='/projetos/:projectId/:experimentId'>
          <ProjectContent />
        </Route>
        {/* implanted experiments content */}
        <Route exact path='/fluxos-implantados'>
          <ImplantedExperimentsContent />
        </Route>
        {/* task content */}
        <Route exact path='/tarefas'>
          <TaskContent />
        </Route>
        {/* error 404 content */}
        <Route path='*'>
          <Error404Content />
        </Route>
      </Switch>
    </Layout>
  );
};

// EXPORT
export default Content;
