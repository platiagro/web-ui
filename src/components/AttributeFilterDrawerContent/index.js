import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Select } from 'antd';
import _ from 'lodash';

import ResultsDrawer from '../ResultsDrawer';
import ResultsButtonBar from '../ResultsButtonBar';

const { Option } = Select;

const AttributeFilterDrawerContent = ({
  dataSets,
  parameter,
  setFilter,
  runStatus,
  taskStatus,
  details,
  targetId,
}) => {
  const optionsTarget = _.filter(dataSets, function (o) {
    return o.uuid !== targetId;
  });
  // resultados
  const [results, setResults] = useState(
    (runStatus === 'Failed' || runStatus === 'Succeeded') &&
    taskStatus === 'Succeeded'
  );

  const [showResults, setShowResults] = useState(results);
  console.log(targetId);

  return (
    <div>
      {!showResults ? (
        <div>
          <p>Atributos excluídos</p>
          <p>
            <small>Selecione os atributos que serão excluídos.</small>
          </p>
          <Select
            onChange={setFilter}
            value={parameter}
            mode='multiple'
            style={{ width: '100%' }}
            placeholder='Selecione'
          >
            {optionsTarget.map((item) => (
              <Option key={item.uuid} value={item.name}>
                {item.name}
              </Option>
            ))}
          </Select>
        </div>
      ) : (
          <ResultsDrawer attributesFilter={parameter} />
        )}
      {runStatus === 'Failed' && taskStatus === 'Succeeded' ? (
        <ResultsButtonBar
          setShowResults={setShowResults}
          showResults={showResults}
        />
      ) : null}
    </div>
  );
};

AttributeFilterDrawerContent.propTypes = {
  dataSets: PropTypes.arrayOf(PropTypes.any).isRequired,
  parameter: PropTypes.arrayOf(PropTypes.string).isRequired,
  setFilter: PropTypes.func.isRequired,
};

export default AttributeFilterDrawerContent;
