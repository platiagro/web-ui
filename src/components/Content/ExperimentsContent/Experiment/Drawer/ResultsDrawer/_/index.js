// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';

// UI LIBS
import { LoadingOutlined } from '@ant-design/icons';
import { Empty, Spin, Tabs } from 'antd';

// COMPONENTS
import { CommonTable } from 'components';
import TagResult from '../TagResult';
import TableResult from '../TableResult';
import PlotResult from '../PlotResult';
import MetricsTitle from './MetricsTitle';

// CONTAINERS
import { DownloadOperatorDatasetContainer } from 'containers';

// STYLES
import './ResultsDrawer.less';

// DESTRUCTURING TABS
const { TabPane } = Tabs;

// RESULTS TYPES
const resultsTypes = {
  // tag
  tag: ({ uuid, ...props }) => <TagResult key={uuid} {...props} />,
  // table
  table: ({ uuid, ...props }, onPageChange) => (
    <TableResult key={uuid} onPageChange={onPageChange} {...props} />
  ),
  // plot
  plot: ({ uuid, ...props }) => <PlotResult key={uuid} {...props} />,
};

/**
 * Results Drawer.
 * This component is responsible for displaying drawer with results.
 *
 * @param {object} props Component props
 * @returns {ResultsDrawer} React component
 */
const ResultsDrawer = (props) => {
  const {
    isToShowDownloadButtons,
    loading,
    metrics,
    metricsLoading,
    onDatasetPageChange,
    parameters,
    results,
    resultsTabStyle,
    scroll,
  } = props;
  // metrics data source
  const dataSource = metrics.map((element, i) => {
    const objectKey = Object.keys(element)[0];
    const objectValor = element[objectKey];
    const obj = {
      key: i,
      metrica: objectKey,
      valor: JSON.stringify(objectValor),
    };
    return obj;
  });

  // metrics columns
  const columns = [
    {
      title: 'Métrica',
      dataIndex: 'metrica',
      key: 'metrica',
      render: (val) => <span style={{ fontWeight: 'bold' }}>{val}</span>,
    },
    {
      title: 'Valor',
      dataIndex: 'valor',
      key: 'valor',
      render: (val) => <span style={{ fontFamily: 'monospace' }}>{val}</span>,
    },
  ];

  // parameters columns
  const parametersColumns = [
    {
      title: 'Parâmetro',
      dataIndex: 'name',
      key: 'parameter',
      render: (val) => <span style={{ fontWeight: 'bold' }}>{val}</span>,
    },
    {
      title: 'Valor',
      dataIndex: 'value',
      key: 'value',
      render: (val) => <span style={{ fontFamily: 'monospace' }}>{val}</span>,
    },
  ];

  return (
    <div className='resultsDrawer'>
      {loading ? (
        <Spin indicator={<LoadingOutlined />} />
      ) : results.length > 0 || metrics.length > 0 || parameters.length > 0 ? (
        <>
          <Tabs defaultActiveKey='1'>
            {/* results */}
            <TabPane tab='Resultados' key='1'>
              <div style={resultsTabStyle}>
                {results.map((result) => (
                  <div className='tab-content' key={result.uuid}>
                    {resultsTypes[result.type](result, onDatasetPageChange)}
                  </div>
                ))}
              </div>
              {isToShowDownloadButtons ? (
                <DownloadOperatorDatasetContainer />
              ) : null}
            </TabPane>

            {/* metrics */}
            <TabPane
              tab={<MetricsTitle loading={metricsLoading} />}
              key='2'
              disabled={metrics.length <= 0}
            >
              <CommonTable
                bordered
                columns={columns}
                dataSource={dataSource}
                isLoading={false}
                rowKey={() => {
                  return uuidv4();
                }}
                scroll={scroll}
              />
            </TabPane>

            {/* parameters */}
            <TabPane
              tab={<span>Parâmetros</span>}
              key='3'
              disabled={parameters.length <= 0}
            >
              <CommonTable
                bordered
                columns={parametersColumns}
                dataSource={parameters}
                isLoading={false}
                rowKey={() => {
                  return uuidv4();
                }}
                scroll={scroll}
              />
            </TabPane>
          </Tabs>
        </>
      ) : (
        <Empty
          image={Empty.PRESENTED_IMAGE_SIMPLE}
          description='Não existem resultados!'
        />
      )}
    </div>
  );
};

// PROP TYPES
ResultsDrawer.propTypes = {
  /** Show download buttons */
  isToShowDownloadButtons: PropTypes.bool,
  /** Results drawer is loading */
  loading: PropTypes.bool.isRequired,
  /** Metrics list */
  metrics: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** Metrics is loading */
  metricsLoading: PropTypes.bool.isRequired,
  /** Training parameters */
  parameters: PropTypes.arrayOf(PropTypes.object),
  /** Results list */
  results: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** CSS style for results tab */
  resultsTabStyle: PropTypes.object,
  /** Table scroll config */
  scroll: PropTypes.object,
};

ResultsDrawer.defaultProps = {
  /** Table scroll config */
  scroll: undefined,
};

// EXPORT
export default ResultsDrawer;
