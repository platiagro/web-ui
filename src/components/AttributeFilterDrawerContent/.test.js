import React from 'react'

import { shallow, mount } from 'enzyme'

import { Select } from 'antd';
import _ from 'lodash';

import AttributeFilterDrawerContent from '.';
import ResultsDrawer from '../ResultsDrawer';
import ResultsButtonBar from '../ResultsButtonBar';

const { Option } = Select;

const setFilterFunc = (e) => {
};

describe('AttributeFilterDrawerContent component', () => {

  it('is expected render without crashing', () => {
    shallow(<AttributeFilterDrawerContent
      parameter={['{"uuid":"uuid", "name":"name"}']}
      dataSets={['']}
      setFilter={setFilterFunc}
    />);
  });

  it('is expected to be of type Div', () => {
    const wrapper = shallow(<AttributeFilterDrawerContent
      parameter={['']}
      dataSets={['']}
      setFilter={setFilterFunc}
    />);

    expect(wrapper.is('div')).toBeTruthy();
  });

  it('Div child is expected to have a Select child', () => {
    const wrapper = shallow(<AttributeFilterDrawerContent
      parameter={['']}
      dataSets={['']}
      setFilter={setFilterFunc}
    />);

    expect(
      wrapper
        .children('div')
        .children(Select)
        .exists()
    ).toBeTruthy();
  });

  it('Select child is expected to have a Option child', () => {
    const wrapper = shallow(<AttributeFilterDrawerContent
      parameter={['']}
      dataSets={[{ "uuid": "uuid", "name": "name" }]}
      setFilter={setFilterFunc}
    />);

    expect(
      wrapper
        .children('div')
        .children(Select)
        .children(Option)
        .exists()
    ).toBeTruthy();
  });

  it('Div child is expected to have a ResultsDrawer child and ResultsButtonBar button', () => {
    const wrapper = mount(<AttributeFilterDrawerContent
      parameter={['']}
      dataSets={['']}
      setFilter={setFilterFunc}
      runStatus={'Failed'}
      taskStatus={'Succeeded'}
    />);

    expect(
      wrapper
        .children('div')
        .children(ResultsDrawer)
        .exists()
    ).toBeTruthy();

    expect(
      wrapper
        .children('div')
        .children(ResultsButtonBar)
        .exists()
    ).toBeTruthy();
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(<AttributeFilterDrawerContent
      parameter={['']}
      dataSets={['']}
      setFilter={setFilterFunc}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
