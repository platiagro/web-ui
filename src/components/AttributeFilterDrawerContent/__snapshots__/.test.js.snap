// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeFilterDrawerContent component is expected render html correctly 1`] = `
<div>
  <div>
    <p>
      Atributos excluídos
    </p>
    <p>
      <small>
        Selecione os atributos que serão excluídos.
      </small>
    </p>
    <Select
      choiceTransitionName="zoom"
      mode="multiple"
      onChange={[Function]}
      placeholder="Selecione"
      showSearch={false}
      style={
        Object {
          "width": "100%",
        }
      }
      transitionName="slide-up"
      value={
        Array [
          "",
        ]
      }
    />
  </div>
</div>
`;
